gui/fitpanelv7/src/RFitPanel6.cxx

```diff
@@ -0,0 +1,512 @@
+      return;
+   }
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+/// Dummy function, called when "Fit" button pressed in UI
+void ROOT::Experimental::RFitPanel6::DrawContour(const std::string &model)
+{
+   // FIXME: do not use static!!!
+  static TGraph * graph = nullptr;
+  std::string options;
+  // TBackCompFitter *fFitter = nullptr;
+  auto obj = TBufferJSON::FromJSON<ROOT::Experimental::RFitPanelModel6>(model);
+
+  if(!obj->fContourImpose) {
+    if(graph){
+      delete graph;
+      options = "ALF";
+      graph= nullptr;
+    }
+  } else {
+    options = "LF";
+  }
+
+
+  if (!graph)
+     graph = new TGraph(static_cast<int>(obj->fContourPoints));
+
+  auto colorid = TColor::GetColor(std::stoi(obj->fColorContour[0]), std::stoi(obj->fColorContour[1]), std::stoi(obj->fColorContour[2]));
+  graph->SetLineColor(colorid);
+
+  if(obj->fContourPar1 == obj->fContourPar2) {
+    Error("DrawContour", "Parameters cannot be the same");
+    return;
+  }
+
+  //fFitter->Contour(obj->fContourPar1, obj->fContourPar2, graph, obj->fConfLevel);
+  //graph->GetXaxis()->SetTitle( fFitter->GetParName(obj->fContourPar1) );
+  //graph->GetYaxis()->SetTitle( fFitter->GetParName(obj->fContourPar2) );
+  //graph->Draw( options.c_str() );
+  gPad->Update();
+
+ //printf("Points %d Contour1 %d Contour2 %d ConfLevel %f\n", obj->fContourPoints, obj->fContourPar1, obj->fContourPar2, obj->fConfLevel);
+}
+
+void ROOT::Experimental::RFitPanel6::DrawScan(const std::string &model)
+{
+
+  auto obj = TBufferJSON::FromJSON<ROOT::Experimental::RFitPanelModel6>(model);
+  static TGraph * graph = nullptr;
+  // TBackCompFitter *fFitter = nullptr;
+
+  // FIXME: do not use static!!!
+  if(graph) {
+     delete graph;
+  }
+  graph = new TGraph(static_cast<int>(obj->fScanPoints));
+  //fFitter->Scan(obj->fScanPar, graph, obj->fScanMin, obj->fScanMax);
+
+  graph->SetLineColor(kBlue);
+  graph->SetLineWidth(2);
+ // graph->GetXaxis()->SetTitle(fFitter->GetParName(obj->fScanPar)); ///???????????
+  graph->GetYaxis()->SetTitle("FCN");
+  graph->Draw("APL");
+  gPad->Update();
+
+
+  //printf("SCAN Points %d, Par %d, Min %d, Max %d\n", obj->fScanPoints, obj->fScanPar, obj->fScanMin, obj->fScanMax);
+}
+
+void ROOT::Experimental::RFitPanel6::DoFit(const std::string &model)
+{
+   // printf("DoFit %s\n", model.c_str());
+   auto obj = TBufferJSON::FromJSON<ROOT::Experimental::RFitPanelModel6>(model);
+
+   ROOT::Math::MinimizerOptions minOption;
+
+   //Fitting Options
+   if (obj) {
+
+      if (gDebug > 0)
+         ::Info("RFitPanel6::DoFit", "range %f %f select %s function %s", obj->fUpdateRange[0], obj->fUpdateRange[1], obj->fSelectDataId.c_str(), obj->fSelectXYId.c_str());
+
+      if (obj->fRealFunc.empty()) {
+         obj->fRealFunc = "gaus";
+      }
+
+      if(!obj->fMinLibrary.empty()){
+        minOption.SetMinimizerAlgorithm(obj->fMinLibrary.c_str());
+      }
+
+      if(!obj->fErrorDef == 0) {
```

Comment:
That is confusing to read and might not do what you expect. You probably simply meant `if (obj->fErrorDef) {`.
More occurrences below.